packages/core/contracts/test/ValidationsNatspec.sol

```diff
@@ -0,0 +1,227 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+contract HasEmptyConstructorNatspec {
+  constructor() { }
+}
+
+/// @custom:openzeppelin-upgrade-allow-unsafe constructor
+contract HasNonEmptyConstructorNatspec1 {
+  constructor() { msg.sender; }
+}
+
+contract HasNonEmptyConstructorNatspec2 {
+  /// @custom:openzeppelin-upgrade-allow-unsafe constructor
+  constructor() { msg.sender; }
+}
+
+contract ParentHasNonEmptyConstructorNatspec1 is HasNonEmptyConstructorNatspec1 {}
+contract ParentHasNonEmptyConstructorNatspec2 is HasNonEmptyConstructorNatspec2 {}
+contract AncestorHasNonEmptyConstructorNatspec1 is ParentHasNonEmptyConstructorNatspec1 {}
+contract AncestorHasNonEmptyConstructorNatspec2 is ParentHasNonEmptyConstructorNatspec2 {}
+
+/// @custom:openzeppelin-upgrade-allow-unsafe state-variable-assignment
+contract HasStateVariableAssignmentNatspec1 {
+  uint x = 1;
+}
+
+contract HasStateVariableAssignmentNatspec2 {
+  /// @custom:openzeppelin-upgrade-allow-unsafe state-variable-assignment
+  uint x = 1;
+}
```

Comment:
I'd like to see a test that something like the following fails:

```solidity
contract HasStateVariableAssignmentNatspec2 {
  /// @custom:openzeppelin-upgrade-allow-unsafe state-variable-assignment
  uint x = 1;

  uint y = 2;
}
```

Consider adding a few more tests like this that should fail to validate, for other overrides. No need to be exhaustive, just a few strategically chosen ones.